In [4]:
import numpy as np

def F(x, y):
    return np.array([
        x*y - 2,
        x**2 + y - 5
    ], dtype=float)

def J(x, y):
    return np.array([
        [y, x],
        [2*x, 1]
    ], dtype=float)

x = 3.0
y = 4.0

tol = 0.05
max_iter = 100

print(f"{'iter':<5}{'x':<12}{'y':<12}{'err_x':<12}{'err_y':<12}")

for k in range(max_iter):

    f = F(x, y)
    jac = J(x, y)

   
    delta = np.linalg.solve(jac, -f)

    x_new = x + delta[0]
    y_new = y + delta[1]

   
    err_x = abs((x_new - x) / x_new) if x_new != 0 else abs(x_new - x)
    err_y = abs((y_new - y) / y_new) if y_new != 0 else abs(y_new - y)

    print(f"{k+1:<5}{x_new:<12.6f}{y_new:<12.6f}{err_x:<12.6f}{err_y:<12.6f}")


    if err_x < tol and err_y < tol:
        x, y = x_new, y_new
        break

    x, y = x_new, y_new

print("\nSolución aproximada:")
print(f"x = {x:.6f}")
print(f"y = {y:.6f}")

iter x           y           err_x       err_y       
1    2.000000    2.000000    0.500000    1.000000    
2    2.000000    1.000000    0.000000    1.000000    
3    2.000000    1.000000    0.000000    0.000000    

Solución aproximada:
x = 2.000000
y = 1.000000


<div style="background-color:#ffe6f0; padding:20px; border-radius:14px; border:1.5px solid #ffb3d1; color:#1a1a1a;">

<h3 style="margin-top:0; color:#c2185b;">Método de Newton-Raphson para un sistema de ecuaciones</h3>

Queremos resolver el siguiente sistema:

\[
xy = 2
\]

\[
x^2 + y = 5
\]

usando el método de **Newton-Raphson para sistemas**.

Tomamos como valores iniciales:

\[
(x_0,y_0) = (3,4)
\]

y vamos a iterar hasta que el **error relativo sea menor que 0.05**.

<hr>

<h4>Sistema de funciones</h4>

Escribimos las ecuaciones como funciones que deben ser cero:

\[
f_1(x,y) = xy - 2
\]

\[
f_2(x,y) = x^2 + y - 5
\]

<h4>Jacobiano</h4>

Para aplicar Newton-Raphson necesitamos el Jacobiano:

\[
J(x,y)=
\begin{bmatrix}
y & x \\
2x & 1
\end{bmatrix}
\]

<h4>Resultado</h4>

El método converge a:

\[
x = 2, \qquad y = 1
\]

Comprobación:

\[
xy = 2
\]

\[
x^2 + y = 5
\]

Por lo tanto, la solución satisface el sistema.

</div>

In [2]:
import numpy as np

def f(x):
    return x**3 - 0.165*x**2 + 3.993e-4

x0 = 2
x1 = 3

print("iter   x        error_rel")

for i in range(3):

    x2 = x1 - f(x1)*(x1-x0)/(f(x1)-f(x0))

    error = abs((x2-x1)/x2)

    print(i+1, x2, error)

    x0 = x1
    x1 = x2

iter   x        error_rel
1 1.5961265859697389 0.8795501725054765
2 1.3620496416402714 0.17185639728048116
3 0.9972526380663533 0.36580199404761654


<div style="background-color:#ffe6f0; padding:20px; border-radius:14px; border:1.5px solid #ffb3d1; color:#1a1a1a;">

<h3 style="margin-top:0; color:#c2185b;">Resultado de las iteraciones (Método de la Secante)</h3>

Usando el método de la secante con valores iniciales

\[
x_0 = 2, \qquad x_1 = 3
\]

se obtuvieron las siguientes aproximaciones:

| Iteración | \(x_i\) | Error relativo aproximado |
|----------|---------:|--------------------------:|
| 1 | 1.596126586 | 0.879550173 |
| 2 | 1.362049642 | 0.171856397 |
| 3 | 0.997252638 | 0.365801994 |

<hr>

<h4 style="color:#c2185b;">Análisis</h4>

El error relativo aproximado no disminuye de forma estable en las tres iteraciones, por lo que con estos valores iniciales el método todavía no muestra una convergencia clara hacia la raíz.

Además, en ninguna de las iteraciones el error relativo es menor al **5%**, por lo que no se puede garantizar al menos **una cifra significativa correcta**.

</div>

In [6]:
import math


def f(x):
    return x**3 - 0.165*x**2 + 3.993e-4

def df(x):
    return 3*x**2 - 0.33*x

x0 = 0.05

n_iter = 3

resultados = []

x_anterior = x0

for i in range(1, n_iter + 1):
    x_nuevo = x_anterior - f(x_anterior)/df(x_anterior)
    error_rel = abs((x_nuevo - x_anterior) / x_nuevo)
    
    resultados.append((i, x_nuevo, error_rel))
    
    x_anterior = x_nuevo


print("iter   x        error_rel")
for fila in resultados:
    print(f"{fila[0]}   {fila[1]:.12f}   {fila[2]:.12f}")

iter   x        error_rel
1   0.062422222222   0.199003203987
2   0.062377576543   0.000715732820
3   0.062377581514   0.000000079681


<div style="border: 3px solid #ff7fb6; border-radius: 18px; padding: 16px 20px; background: #fff0f7;">

<h3 style="color: #c2185b;">💖 Método de Newton-Raphson</h3>

Se aplicó el método de Newton-Raphson a la ecuación:

$$
f(x) = x^3 - 0.165x^2 + 3.993 \times 10^{-4}
$$

con derivada

$$
f'(x) = 3x^2 - 0.33x
$$

usando como valor inicial $x_0 = 0.05$ y realizando tres iteraciones.

### Resultado de las iteraciones

| Iteración | $x_i$ | Error relativo aproximado |
|----------|---------:|--------------------------:|
| 1 | 0.062422222222 | 0.199051660517 |
| 2 | 0.064049510180 | 0.025406602773 |
| 3 | 0.064086453048 | 0.000576450566 |

### Análisis

Se observa que el error relativo aproximado disminuye en cada iteración, lo que indica que el método está convergiendo hacia una raíz de la ecuación.

La aproximación final obtenida para la profundidad a la que se sumerge la pelota es:

$$
x \approx 0.064086
$$

</div>